lib/chef/provider/aws_cloudsearch_domain.rb

```diff
@@ -127,8 +127,10 @@ def create_index_field(field)
   end
 
   def create_index_fields
-    new_resource.index_fields.each do |field|
-      create_index_field(field)
+    if ! new_resource.index_fields.nil?
```

Comment:
Can you make this `unless` instead of `if !` please?  Then I'll accept this PR - I've got the tests running locally and I'll work on fixing travis while merging some more of these PRs